### DISCUSSION TITLE
Can you implement an operator that works on a series of values instead of just a single scalar, such as the fft?

### DISCUSSION THREAD

**Discussion by mchiquier** [MilesCranmer/PySR#476] [General]
Can you implement an operator that works on a series of values instead of just a single scalar, such as the fft?

**Comment by MilesCranmer** [comment]
Not yet, unfortunately. You could always write a custom evaluation routine in a passed loss function, but it will be difficult. Maybe if it is just a single FFT in a fixed position of the expression it wouldn't be so bad to add.

The way this will eventually be added (for very flexible expressions) is by allowing non-scalar equations. You can see at the bottom of the README on https://github.com/SymbolicML/DynamicExpressions.jl/, the backbone of SymbolicRegression.jl can already handle these sorts of things. It's just it needs to be set up for the rest of the code.

**Reply by mchiquier** [reply]
Great, thanks!! This is exciting. Do you have a rough timeline estimate for when this will be set up for the rest of the code?

**Reply by MilesCranmer** [reply]
If I do it, it may take 1 year or more unfortunately – only because busy with teaching/etc. If I can find an interested student to take it on, it could probably be done much quicker

**Reply by mchiquier** [reply]
I see - thanks!
